lib/pages/caregiver/awards_page.dart

```diff
@@ -28,68 +30,92 @@ class _CaregiverAwardsPageState extends State<CaregiverAwardsPage> {
+						HeaderActionButton.normal(Icons.add, '$_pageKey.header.addReward', 
+						() => Navigator.of(context).pushNamed(AppPage.caregiverRewardForm.name, arguments: AppFormType.create)),
 						HeaderActionButton.normal(Icons.add, '$_pageKey.header.addBadge', 
 						() => Navigator.of(context).pushNamed(AppPage.caregiverBadgeForm.name, arguments: AppFormType.create))
 					]),
-					AppSegments(
-						segments: [
-							Segment(
-								title: '$_pageKey.content.addedAwardsTitle',
-								noElementsMessage: '$_pageKey.content.noAwardsAdded',
-								noElementsAction: RaisedButton(
-									child: Text(
-										AppLocales.of(context).translate('$_pageKey.header.addAward'),
-										style: Theme.of(context).textTheme.button
-									),
-									onPressed: () => {},
-								),
-								elements: <Widget>[
-									ItemCard(
-										title: "Wycieczka do Zoo", 
-										subtitle: AppLocales.of(context).translate('$_pageKey.content.limitedAward', {'AWARD_LIMIT': 1}),
-										menuItems: [
-											UIButton.ofType(ButtonType.edit, () => {log("edit")}),
-											UIButton.ofType(ButtonType.delete, () => {log("delete")})
-										],
-										graphicType: GraphicAssetType.awardsIcons,
-										graphic: 16,
-										chips: <Widget>[
-											AttributeChip.withCurrency(content: "30", currencyType: CurrencyType.diamond, tooltip: '$_pageKey.content.pointCost')
-										],
-									)
-								]
-							),
-							Segment(
-								title: '$_pageKey.content.addedBadgesTitle',
-								noElementsMessage: '$_pageKey.content.noBadgesAdded',
-								noElementsAction: RaisedButton(
-									child: Text(
-										AppLocales.of(context).translate('$_pageKey.header.addBadge'),
-										style: Theme.of(context).textTheme.button
-									),
-									onPressed: () => {},
-								),
-								elements: <Widget>[
-									ItemCard(
-										title: "Super planista", 
-										subtitle: AppLocales.of(context).translate('$_pageKey.content.3LeveledBadge'),
-										menuItems: [
-											UIButton.ofType(ButtonType.edit, () => {log("edit")}),
-											UIButton.ofType(ButtonType.delete, () => {log("delete")})
-										],
-										graphicType: GraphicAssetType.badgeIcons,
-										graphic: 3,
-										graphicHeight: 44.0,
-									)
-								]
-							)
-						]
+					LoadableBlocBuilder<CaregiverAwardsCubit>(
+						builder: (context, state) => AppSegments(segments: _buildPanelSegments(state, context))
 					)
 				]
 			),
 			bottomNavigationBar: AppNavigationBar.caregiverPage(currentIndex: 2)
     );
 	}
+	
+	String _getBadgeMaxLevel(UIBadgeMaxLevel maxLevel) {
+		String _textKey;
+		switch(maxLevel) {
+		  case UIBadgeMaxLevel.one:
+		    _textKey = 'noLeveledBadge';
+		    break;
+		  case UIBadgeMaxLevel.three:
+		    _textKey = '3LeveledBadge';
+		    break;
+		  case UIBadgeMaxLevel.five:
+		    _textKey = '5LeveledBadge';
```

Comment:
This extension could than be applied here to create the key and get rid of conditions (provided you replace `no` with `1`)